src/app/utils/di-tokens.ts

```diff
@@ -0,0 +1,6 @@
+// Following this post to allow testing of the location object: https://itnext.io/testing-browser-window-location-in-angular-application-e4e8388508ff
+import { InjectionToken } from "@angular/core";
+
+export const LOCATION_TOKEN = new InjectionToken<Location>(
```

Comment:
somehow feels wrong to have this in an isolated, super generic "utils". But I guess it is used for multiple unrelated things?

Reply:
It is kind utils in a way that this only makes things easier to test but doesn't add any functionality. The reason I moved it out was also due to circular dependency issues between files.